lib/Ohmbrewer_Thermostat.cpp

```diff
@@ -176,11 +78,94 @@ Ohmbrewer::Thermostat::~Thermostat() {
 }
 
 /**
+ * logic for initializing the constructors
+ */
+void Ohmbrewer::Thermostat::initThermostat(int id, std::list<int>* thermPins){
```

Comment:
Better, but this is still pretty hairy. I'd split it into initPID and initPins...